physics/module_sf_noahmplsm.F90

```diff
@@ -5382,160 +5501,431 @@ subroutine sfcdif3(parameters,iloc    ,jloc    ,iter    ,sfctmp  ,qair    ,ur
     tem2   = max(fveg, 0.1_kind_phys)
     zvfun1 = sqrt(tem1 * tem2)
     gdx    = sqrt(garea1)
+    
+    gdx    = 3000.0   ! this will remove gdx effect
+    zvfun1 = 1.0      ! this will remove zvfun effect
 
     if(thsfc_loc) then            ! Use local potential temperature
       tvs   = tgb * virtfac
     else                          ! Use potential temperature referenced to 1000 hPa
       tvs   = tgb/prsik1x * virtfac
     endif
 
-    call stability (zlvlb, zvfun1, gdx, tv1, thv1, ur, z0m, z0h, tvs, grav, thsfc_loc,  &
+    call gfs_stability (zlvlb, zvfun1, gdx, tv1, thv1, ur, z0m, z0h, tvs, grav, thsfc_loc,  &
          rb1, fm,fh,fm10,fh2,cm,ch,stress1,fv)
 
   end subroutine sfcdif3
 
+!== begin gfs_stability ==================================================================================
```

Comment:
Can you say something about why this is being duplicated here from sfc_diff.f?

Reply:
The goal for us here is to remove dependency on other CCPP modules inside noahmp so that we can contribute back to the community code and eventually have a tighter relationship with the noahmp community repository.